x-pack/packages/kbn-elastic-assistant/impl/assistant/api/conversations/use_bulk_actions_conversations.ts

```diff
@@ -0,0 +1,152 @@
+  message: string;
+  status_code: number;
+  err_code?: string;
+  conversations: Array<{ id: string; name?: string }>;
+}
+
+export interface BulkActionAttributes {
+  summary: BulkActionSummary;
+  results: BulkActionResult;
+  errors?: BulkActionAggregatedError[];
+}
+
+export interface BulkActionResponse {
+  success?: boolean;
+  conversations_count?: number;
+  message?: string;
+  statusCode?: number;
+  attributes: BulkActionAttributes;
+}
+
+export interface ConversationUpdateParams {
+  id?: string;
+  title?: string;
+  messages?: Message[];
+  apiConfig?: {
+    connectorId?: string;
+    connectorTypeTitle?: string;
+    defaultSystemPromptId?: string;
+    provider?: OpenAiProviderType;
+    model?: string;
+  };
+}
+
+export interface ConversationsBulkActions {
+  update?: Record<string, ConversationUpdateParams>;
+  create?: Record<string, Conversation>;
+  delete?: {
+    ids: string[];
+  };
+}
+
+const transformCreateActions = (
+  createActions: Record<string, Conversation>,
+  conversationIdsToDelete?: string[]
+) =>
+  Object.keys(createActions).reduce((conversationsToCreate: Conversation[], conversationId) => {
+    if (createActions && !conversationIdsToDelete?.includes(conversationId)) {
+      conversationsToCreate.push(createActions[conversationId]);
+    }
+    return conversationsToCreate;
+  }, []);
+
+const transformUpdateActions = (
+  updateActions: Record<string, ConversationUpdateParams>,
+  conversationIdsToDelete?: string[]
+) =>
+  Object.keys(updateActions).reduce(
+    (conversationsToUpdate: ConversationUpdateParams[], conversationId) => {
+      if (updateActions && !conversationIdsToDelete?.includes(conversationId)) {
+        conversationsToUpdate.push({
+          id: conversationId,
+          ...updateActions[conversationId],
+        });
+      }
+      return conversationsToUpdate;
+    },
+    []
+  );
+
+export const bulkChangeConversations = async (
+  http: HttpSetup,
+  conversationsActions: ConversationsBulkActions,
+  toasts?: IToasts
+) => {
+  // transform conversations disctionary to array of Conversations to create
+  // filter marked as deleted
+  const conversationsToCreate = conversationsActions.create
+    ? transformCreateActions(conversationsActions.create, conversationsActions.delete?.ids)
+    : undefined;
+
+  // transform conversations disctionary to array of Conversations to update
+  // filter marked as deleted
+  const conversationsToUpdate = conversationsActions.update
+    ? transformUpdateActions(conversationsActions.update, conversationsActions.delete?.ids)
+    : undefined;
+
+  try {
+    const result = await http.fetch<BulkActionResponse>(
+      ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL_BULK_ACTION,
+      {
+        method: 'POST',
+        version: ELASTIC_AI_ASSISTANT_API_CURRENT_VERSION,
+        body: JSON.stringify({
+          update: conversationsToUpdate,
+          create: conversationsToCreate,
+          delete: conversationsActions.delete,
+        }),
```

Comment:
nit: Can short circuit this request if body is empty `{}`. This can be seen when entering `Settings` and changing KB settings and clicking `Save` -- this request is still made even though no conversation changes occurred.

Reply:
Solved by https://github.com/elastic/kibana/pull/173487/commits/d39d169a3a89c81d361313be69463e8fd13b8804